ts/main.ts

```diff
@@ -1,66 +1,326 @@
+        text: 'WATER',
+      },
+      {
+        checkFor: 'frameType',
+        val: 'normal',
+        imgPath: './images/normal.png',
+        text: 'Normal',
+      },
+      {
+        checkFor: 'race',
+        val: 'Warrior',
+        imgPath: './images/warrior.png',
+        text: 'Warrior',
+      },
+    ],
+  },
+  '2': {
+    topCategories: [
+      {
+        checkFor: 'race',
+        val: 'Spellcaster',
+        imgPath: './images/spellcaster.png',
+        text: 'Spellcaster',
+      },
+      { checkFor: 'ATK', val: 1800, imgPath: './images', text: '1800 ATK' },
+      {
+        checkFor: 'atk',
+        val: 2800,
+        imgPath: './images/ATK.png',
+        text: '2800 ATK',
+      },
+    ],
+    leftCategories: [
+      {
+        checkFor: 'attribute',
+        val: 'FIRE',
+        imgPath: './images/FIRE.png',
+        text: 'FIRE',
+      },
+      { checkFor: 'level', val: 7, imgPath: './images', text: 'Level 7' },
+      {
+        checkFor: 'frameType',
+        val: 'synchro',
+        imgPath: './images/synchro.png',
+        text: 'Synchro',
+      },
+    ],
+  },
+};
+
+const currentGridId: number = 1;
+const gridSize: number = 3;
+let numCorrect: number = 0;
+
+const $topCategories = document.querySelectorAll(
+  '.top-cat',
+) as NodeListOf<HTMLDivElement>;
+if (!$topCategories) throw new Error('.top-cat query failed!');
+const $leftCategories = document.querySelectorAll(
+  '.left-cat',
+) as NodeListOf<HTMLDivElement>;
+if (!$leftCategories) throw new Error('left-cat query failed!');
+
+function renderGridCategories(gridID: number): void {
+  for (let i = 0; i < gridSize; i++) {
+    const imgContainerTop: HTMLDivElement = document.createElement('div');
+    imgContainerTop.classList.add('w-1/2', 'h-1/2', 'flex', 'justify-center');
+    const catImgTop: HTMLImageElement = document.createElement('img');
+    catImgTop.setAttribute(
+      'src',
+      categorySets[gridID].topCategories[i].imgPath,
+    );
+    imgContainerTop.appendChild(catImgTop);
+    $topCategories[i].appendChild(imgContainerTop);
+    const catText: HTMLHeadingElement = document.createElement('h1');
+    catText.textContent = categorySets[gridID].topCategories[i].text;
+    $topCategories[i].appendChild(catText);
+
+    const imgContainerLeft: HTMLDivElement = document.createElement('div');
+    imgContainerLeft.classList.add('w-1/2', 'h-1/2', 'flex', 'justify-center');
+    const catImgLeft: HTMLImageElement = document.createElement('img');
+    catImgLeft.setAttribute(
+      'src',
+      categorySets[gridID].leftCategories[i].imgPath,
+    );
+    imgContainerLeft.appendChild(catImgLeft);
+    $leftCategories[i].appendChild(imgContainerLeft);
+    const catTextLeft = document.createElement('h1');
+    catTextLeft.textContent = categorySets[gridID].leftCategories[i].text;
+    $leftCategories[i].appendChild(catTextLeft);
+  }
+} // function that renders Grid Categories based on what number Grid
```

Comment:
Do we need to use the DOM to create the categories and append them, or could we potentially have them already with the grid and then just update the values after shuffling around the categories?

Reply:
I just figured that would be the best way to go about it. In the future, users would be switching through pre-made playable sets of grid categories stored in an object. There might be a case where they want to switch to a set of categories then switch back to it so I would render them using the object with all of the sets of categories and the DOM to create them